api/utils/helper.py

```diff
@@ -16,3 +16,16 @@ def serialize_point(point):
     serialized_point["owner"] = point.user.name
 
     return serialized_point
+
+
+def get_status_code(dictionary):
+    """Get the status code value in a nested dictionary."""
+    status_code_key = 'status_code'
+    for key, value in dictionary.items():
+        if status_code_key == key:
+            status_code_value = value
+        elif isinstance(value, dict):
+            for result in get_status_code(value):
+                status_code_value = result
+
+    yield status_code_value
```

Comment:
@jonathankamau Can you please provide more context on this?

Reply:
@georgreen the `status code` for each type of response is retrieved from the `marshmallow schema` together with the corresponding `message`. Depending on the kind of validation whether it is by a `field validator` or `custom method validator` the `status code` key can either appear in a dictionary or in a nested one. This function searches for all instances of the status code key, which there can only be one, and returns the status code.